### PR TITLE
feat: keep core features supported when skipping extended tests

### LINKED ISSUE #3510: Ensure core features aren't omitted if any extended features are removed
*Description*:
>Describe the desired behavior, what scenario it enables and how it
would be used.

Update the logic of supported features to ensure that core features aren't omitted if any extended features are removed.

[optional *Relevant Links*:]
>Any extra documentation required to understand the issue.

Following https://github.com/kubernetes-sigs/gateway-api/issues/3097#issuecomment-2136085787

### PR DESCRIPTION
**Pull request by levikobi**
**What type of PR is this?**

feat: keep core features supported when skipping extended tests

**Which issue(s) this PR fixes**:

Fixes #3510

### CODE REVIEW

File: internal/gatewayapi/status/gatewayclass.go [hunk 1]
```diff
@@ -76,8 +77,21 @@ func getSupportedFeatures(gatewaySuite suite.ConformanceOptions, skippedTests []
...
+				supportedFeatures.Delete(feature)
+			}
+		case conformance.Extended:
+			var extendedFeatures []features.SupportedFeature
+			for _, feature := range skippedTest.Features {
+				if conformance.GetFeatureSupportLevel(feature) == conformance.Extended {
+					extendedFeatures = append(extendedFeatures, feature)
+				}
+			}
+			for _, feature := range extendedFeatures {
```
Comment (arkodg) [inline internal/gatewayapi/status/gatewayclass.go:92]: should this be out of the `skippedTest` for loop ?

Comment (levikobi) [reply internal/gatewayapi/status/gatewayclass.go]: Hi @arkodg
I updated the function, let me know if it's ok

**Review by arkodg — APPROVED** [review]
LGTM thanks !

**Review by zirain — APPROVED** [review]

**Review by shawnh2 — APPROVED** [review]

**Pull request merged**